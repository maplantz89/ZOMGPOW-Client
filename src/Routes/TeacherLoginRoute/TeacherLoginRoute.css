.teacher-login{
  font-family: var(--heading-font);
}

