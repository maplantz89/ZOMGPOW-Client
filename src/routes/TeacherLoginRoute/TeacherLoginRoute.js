import React, { Component } from 'react';
import TeacherLoginForm from '../../Components/TeacherLoginForm/TeacherLoginForm';
import config from '../../config';
import './TeacherLoginRoute.css';

class TeacherLoginRoute extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => { },
    },
  }
 
  handleLoginSuccess = () => {
    this.props.handleHeaderType(config.TEACHER);
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/dashboard/teacher'
    history.push(destination)
  }

  render() {
    return (
      <section className="teacher-login">
        <TeacherLoginForm
          onLoginSuccess={this.handleLoginSuccess}
        />
      </section>
    );
  }
}

export default TeacherLoginRoute;