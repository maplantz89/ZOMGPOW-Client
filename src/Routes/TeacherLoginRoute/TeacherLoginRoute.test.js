import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TeacherLoginRoute from './TeacherLoginRoute';

describe(`TeacherLoginRoute component`, () => {
  it('renders a .TeacherLoginRoute by default', () => {
    const wrapper = shallow(<TeacherLoginRoute />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})