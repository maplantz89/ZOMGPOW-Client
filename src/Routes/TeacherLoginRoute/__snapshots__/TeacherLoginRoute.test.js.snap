// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherLoginRoute component renders a .TeacherLoginRoute by default 1`] = `
<section>
  <h2>
    Login
  </h2>
  <TeacherLoginForm
    onLoginSuccess={[Function]}
  />
</section>
`;
